src/control/control.cpp

```diff
@@ -42,16 +42,19 @@ ControlDoublePrivate::ControlDoublePrivate(ConfigKey key,
                        Stat::SAMPLE_VARIANCE | Stat::MIN | Stat::MAX),
           m_confirmRequired(false),
           m_pCreatorCO(pCreatorCO) {
-    initialize();
+    initialize(initialValue);
 }
 
-void ControlDoublePrivate::initialize() {
+void ControlDoublePrivate::initialize(double initialValue) {
     double value = 0;
     if (m_bPersistInConfiguration) {
         UserSettingsPointer pConfig = ControlDoublePrivate::s_pUserConfig;
-        if (pConfig != NULL) {
-            // Assume toDouble() returns 0 if conversion fails.
-            value = pConfig->getValueString(m_key).toDouble();
+        if (pConfig != nullptr) {
+            bool conversionWorked;
+            value = pConfig->getValueString(m_key).toDouble(&conversionWorked);
+            if (!conversionWorked) {
+                value = initialValue;
+            }
         }
     }
     m_defaultValue.setValue(0);
```

Comment:
Yeah, that's what I initially did but I wasn't sure if that would always be desirable. I suppose it would be easy enough to override with another call to setDefaultValue() though.